### ISSUE TITLE
Device Migration

### ISSUE THREAD

**Issue by CarloP95** [kubeedge/kubeedge#1372]
**What would you like to be added/modified**:
Support Device Migration scenarios.

**Why is this needed**:
A mobile device could move from a edge node to another one.
For now one must get to the **control plane** (Cloud) and trigger commands manually.

If you are not planning to support migration, it would be good for me to know the motivation.

**Comment by kevin-wangzefeng** [comment]
Thanks @CarloP95 we do have consideration to support device migration when designing the device CRD, that's why a device uses nodeSelector to indicate binding relationship between device and edge nodes. But we not yet provide any complete practise document of using this functionality.

**Comment by CarloP95** [comment]
Hi @kevin-wangzefeng, I'm sorry that I haven't expressed myself well.
Let me describe a use case in which this feature would be useful.

**TL;DR**
In a **5G scenario**, mobile devices can be connected to an edge node but, since they can move between subnets, migration might be necessary.
This migration is not carried automatically. Are you planning to support that?

**Explanation**
Let's suppose that a mobile device is connected with an edge node through a Mapper. At a certain time, this device moves and loses connection with the edge node with which it was deployed (using _nodeSelector_). 

_Let's suppose that before losing connection, the device could send information about the new SSID (or something else that identifies the subnet) to its Mapper, so Controllers on the control plane have all the necessary information to migrate the device instance and Mapper._

It is possible to implement a migration strategy with which the Mapper and/or the DeviceTwin are moved to the new edge node and the device starts working again? 

From my understanding, at the current time one should:
1. **Create a new device** or edit the old device (with _nodeSelector_ set to the new edge node) from the control plane.
2. **Create a new deployment** from the control plane for the Mapper to be deployed into the new edge node (but in this way one can loses state information if the Mapper holds any of these).
3. **Clean up** old device and old deployment from the control plane.

**Comment by qihandeng2** [comment]
> Hi @kevin-wangzefeng, I'm sorry that I haven't expressed myself well.
> Let me describe a use case in which this feature would be useful.
> 
> **TL;DR**
> In a **5G scenario**, mobile devices can be connected to an edge node but, since they can move between subnets, migration might be necessary.
> This migration is not carried automatically. Are you planning to support that?
> 
> **Explanation**
> Let's suppose that a mobile device is connected with an edge node through a Mapper. At a certain time, this device moves and loses connection with the edge node with which it was deployed (using _nodeSelector_).
> 
> _Let's suppose that before losing connection, the device could send information about the new SSID (or something else that identifies the subnet) to its Mapper, so Controllers on the control plane have all the necessary information to migrate the device instance and Mapper._
> 
> It is possible to implement a migration strategy with which the Mapper and/or the DeviceTwin are moved to the new edge node and the device starts working again?
> 
> From my understanding, at the current time one should:
> 
> 1. **Create a new device** or edit the old device (with _nodeSelector_ set to the new edge node) from the control plane.
> 2. **Create a new deployment** from the control plane for the Mapper to be deployed into the new edge node (but in this way one can loses state information if the Mapper holds any of these).
> 3. **Clean up** old device and old deployment from the control plane.

I am interested in this scenario too. There must be one mechanism to keep the running state from source and continue to serve at the new edge node.  
I have several confusions:
1. How to save the running state at the old edge node?
2. How to transfer the saved state? directly between the two edge nodes? or synchronize to cloud and then target to the new edge node?
